dev/fattest.simplicity/src/componenttest/rules/repeater/FeatureSet.java

```diff
@@ -147,16 +183,27 @@ public FeatureSetBuilder(FeatureSet featureSet) {
          * @param eeVersion the EE version these features are based on
          */
         public FeatureSetBuilder(Set<String> features, EEVersion eeVersion) {
-            this.features = new HashSet<>(features);
-            this.eeVersion = eeVersion;
+            this(features, eeVersion, eeVersion.getMinJavaLevel());
         }
 
         /**
          * Create a new builder, initially with no features.
          */
-        public FeatureSetBuilder(EEVersion eeVersion) {
-            this.features = new HashSet<>();
+        public FeatureSetBuilder(EEVersion eeVersion, int minJavaLevel) {
+            this(Collections.emptySet(), eeVersion, minJavaLevel);
+        }
+
+        /**
+         * Create a new builder, starting with the a given set of features
+         *
+         * @param features     a set of features to initially add
+         * @param eeVersion    the EE version these features are based on
+         * @param minJavaLevel the minimum Java SE Version which this FeatureSet can run on.
+         */
+        public FeatureSetBuilder(Set<String> features, EEVersion eeVersion, int minJavaLevel) {
+            this.features = new HashSet<>(features);
             this.eeVersion = eeVersion;
+            this.minJavaLevel = minJavaLevel;
         }
```

Comment:
This builder already has too many constructors.

Reply:
Removed several constructors